README.md

```diff
@@ -1,31 +1,52 @@
 
-## What is CARAMEL?
+## What is Succinct Retrieval?
+
+As defined in [Dillinger, et al. (2022)](https://drops.dagstuhl.de/storage/00lipics/lipics-vol233-sea2022/LIPIcs.SEA.2022.4/LIPIcs.SEA.2022.4.pdf), a *retrieval data structure* $D$ for a static function $f \colon S \to \{0, 1\}^r$ is an index that returns $f(x)$ for any $x \in S$ and any value for $x \notin S$. A highly desirable goal in both theory and practice is to represent $D$ with as little space as possible. To that end, a *succinct retrieval data structure* is an index that uses only $(1 + o(1))r|S|$ bits of space, just slightly more than the information-theoretic lower bound of $r|S|$ bits.
 
-CARAMEL (Compressed Array Representation And More Efficient Lookups) is a data structure built around compressed static functions (CSFs). For a solid introduction to CSFs, see the excellent series of papers by Sebastiano Vigna's group. We draw heavy inspiration from their [Java implementation in Sux4J](https://sux4j.di.unimi.it).
+In the age of massive datasets and machine learning, we observe that numerous applications may benefit from succinct retrieval structures. For example, many industrial search and recommendation systems services reply on computationally expensive machine learning models to compute results. Since executing these queries in real-time is prohibitively expensive, these services often cache the most frequently issues queries along with their corresponding results for faster and more cost-effective serving. As search traffic volume increases over time, it becomes increasingly desirable to store this cache in as few bits as possible via succinct retrieval methods.
 
-**Why caramel?** Compared to Sux4J, we offer about 25% faster construction, a much smaller and simpler codebase (> 10k vs 2k LoC for the core implementation), and easy Python bindings. We support arbitrary data types for keys and values, enabling more flexibility for applications (the original implementation supports only string keys and integer values). We also have some algorithmic improvements to make CSF-backed tables practical for real applications. For example, we have optimal-size Bloom prefilters for situations when one value is very common. We also have value-permutation methods to preprocess 2D arrays into a more compressible format, as well as tricks to implement ragged arrays and other lookup primitives using CSFs. Our paper describes these tricks and applications.
+## A Simple Example
 
-### Minimal API Example
+For a concrete illustration of succinct retrieval in practice, let's take the example of an e-commerce product search engine that wishes to cache pairs of frequently issues search queries and their corresponding results. For instance, a query such as "red shoes" might map to a list of relevant product ids ["A1", "A2", "A3", ...]. We then wish to construct a static look-up table to serve these frequent queries and avoid having to execute the more expensive search engine pipeline each time. 
```

Comment:
Nit: "that wishes to cache" -> "that caches"

Reply:
Changed